lib/srv/desktop/audit.go

```diff
@@ -23,10 +23,43 @@ import (
 	"github.com/gravitational/teleport/api/types"
 	"github.com/gravitational/teleport/api/types/events"
 	libevents "github.com/gravitational/teleport/lib/events"
+	"github.com/gravitational/teleport/lib/generics"
+	"github.com/gravitational/teleport/lib/srv/desktop/tdp"
 	"github.com/gravitational/teleport/lib/tlsca"
 	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
 )
 
+const (
+	unknownName = "unknown"
+)
+
+// sharedDirectoryNameMap keeps a mapping of sessionId
+// to a shared directory's name, for use by the audit log.
+type sharedDirectoryNameMap struct {
+	generics.ThreadSafeMap[string, string]
+	log logrus.FieldLogger
+}
+
+func newSharedDirectoryNameMap(log logrus.FieldLogger) *sharedDirectoryNameMap {
+	return &sharedDirectoryNameMap{
+		ThreadSafeMap: generics.NewThreadSafeMap[string, string](),
+		log:           log,
+	}
+}
+
+func (n *sharedDirectoryNameMap) get(sessionID string) string {
+	var directoryName string
+	if name, ok := n.ThreadSafeMap.Get(sessionID); ok {
+		directoryName = name
+	} else {
+		directoryName = unknownName
+		n.log.Warnf("received a SharedDirectoryAcknowledge event for unknown directory in session: %v", sessionID)
```

Comment:
I don't think this is the right place for this log. I can imagine there may be multiple scenarios where you want to `get` from this map, and not all of them will be in response to a `SharedDirectoryAcknowledge`.